modules/sql/managed-instance/vulnerability-assessment/.bicep/nested_storageRoleAssignment.bicep

```diff
@@ -0,0 +1,19 @@
+@description('Required. The name of the storage account to apply the role assignment to.')
+param storageAccountName string
+
+@description('Required. The principal ID of the SQL managed instance system assigned managed identity to assign the role to.')
+param managedInstancePrincipalId string
+
+resource storageAccount 'Microsoft.Storage/storageAccounts@2021-09-01' existing = {
```

Comment:
Note, if we implement it like this, the storage account must be in the same RG as the SQL MI.

Reply:
This works becuase back in vulnerability-assessment main.bicep, the call to this module sets the scope based on the storage acct resource group, then it passes in stg acct name to allow us just to use existing. I tested on a stg acct in a different resource group and it worked without issue.